File: xabber/src/main/java/com/xabber/android/ui/adapter/chat/BasicMessageVH.java
package com.xabber.android.ui.adapter.chat;

import android.view.View;
import android.widget.TextView;

import androidx.annotation.StyleRes;
import androidx.recyclerview.widget.RecyclerView;

import com.xabber.android.R;

public class BasicMessageVH extends RecyclerView.ViewHolder {

    TextView messageText;
    public boolean needDate;
    public String date;

    BasicMessageVH(View itemView, @StyleRes int appearance) {
        super(itemView);

        messageText = (TextView) itemView.findViewById(R.id.message_text);
        messageText.setTextAppearance(itemView.getContext(), appearance);
    }

    BasicMessageVH(View itemView) {
        super(itemView);

        messageText = (TextView) itemView.findViewById(R.id.message_text);
    }
}

File: xabber/src/main/java/com/xabber/android/ui/adapter/chat/BasicMessageVH.kt
package com.xabber.android.ui.adapter.chat

import android.view.View
import android.widget.TextView
import androidx.annotation.StyleRes
import androidx.appcompat.widget.AppCompatTextView
import androidx.recyclerview.widget.RecyclerView
import com.xabber.android.R
import com.xabber.android.ui.widget.CorrectlyTouchEventTextView

abstract class BasicMessageVH internal constructor(
    itemView: View, @StyleRes appearance: Int
) : RecyclerView.ViewHolder(itemView) {

    val messageText: AppCompatTextView = itemView.findViewById(R.id.message_text)
    var needDate = false
    var date: String? = null

    init {
        messageText.setTextAppearance(itemView.context, appearance)
    }

}